src/main.rs

```diff
@@ -62,6 +63,13 @@ fn interp(prg: &[u8], map: Vec<usize>) {
     let mut ptr = 0;
     let mut cells = Vec::with_capacity(30000);
     cells.resize(30000, 0u8);
+    let stdin = {
+        let mode = CString::new("r").unwrap();
+        let mode_raw = mode.into_raw();
+        let stdin = unsafe { fdopen(libc::STDIN_FILENO, mode_raw) };
+        unsafe { CString::from_raw(mode_raw) };
```

Comment:
What's this line for? It makes a `CString`, but doesn't assign it?

Reply:
No, it actually makes it possible to `free` it.

Reply:
Ohhhh. I see.

Could we use `mode.as_ptr()` instead of `mode.into_raw()`, then Rust retains ownership, and `from_raw` shouldn't be necessary?

(might be wrong)

Reply:
Yes, we can! See the new commit. Good spot.

Reply:
Beautiful!